### PR TITLE
Allow `syndesis-template.yml` to be downloaded from fuse-online-install

### PR DESCRIPTION
**Pull request by lgarciaaco**
Fixes https://github.com/syndesisio/syndesis/issues/5925

### CODE REVIEW

File: install/operator/pkg/syndesis/template/install.go [hunk 1]
```diff
@@ -44,7 +46,25 @@ func GetInstallResourcesAsRuntimeObjects(scheme *runtime.Scheme, syndesis *v1alp
 }
 
 func GetInstallResources(scheme *runtime.Scheme, syndesis *v1alpha1.Syndesis, params InstallParams) ([]runtime.RawExtension, error) {
-	res, err := util.LoadResourceFromFile(scheme, *configuration.TemplateLocation)
+	var templateLocation string
+
+	if releaseVersion := *configuration.ReleaseVersion; len(releaseVersion) > 0 {
+		var fileUrl bytes.Buffer
```
Comment (zregvart) [inline install/operator/pkg/syndesis/template/install.go]: I'd go with `fmt.Sprintf` here instead, something like:
```suggestion
		fileUrl := fmt.Sprintf("https://raw.githubusercontent.com/syndesisio/fuse-online-install/%s/resources/fuse-online-template.yml", releaseVersion)
```

Would also be good to put `"https://raw.githubusercontent.com/syndesisio/fuse-online-install/%s/resources/fuse-online-template.yml"` into a constant.

**Comment by lgarciaaco** [comment]
Yea I agree it is more readable that way

**Comment by lgarciaaco** [comment]
> This is a good first iteration, we need to track this for the next release and make sure that the CRD in the newly re-factored operator has this `releaseVersion` property. Also it might bestow us to serve this file from a syndesis/Fuse/Red Hat related domain instead from raw.githubusercontent.com.

We have discussed about the CRD idea. Having the release version in the CR might be problematic the way it is right now, where we deliver fuse/syndesis as one (all images for different components are tested together). The upgrade process makes sure to upgrade the operator to match the version that is targeted. If we build the version in the CR, we have giving customers the option to blow up the stack by easily changing the version to something the operator doesn't support. We would have to take this into consideration.

**Comment by lgarciaaco** [comment]
> I do wonder what issues we'd encounter if the download fails for instance in firewalled installation of OpenShift without network access. And in this case how is that communicated to the end user.

But isn't this the same constrain we have now? ATM https://raw.githubusercontent.com/syndesisio/fuse-online-install/1.6.21/resources/fuse-online-image-streams.yml is downloaded by `install_ocp.sh`, so an internet connection is required.

Either way, I think the operator should fail the installation if the file can't be downloaded, which is the current behavior

**Comment by nicolaferraro** [comment]
Yeah, so I agree that the `--release-version` flag should only go in the operator, while the CRD already tracks the current version that is installed. The operator makes sure they match at the end of the upgrade process.

In case of download error, the operator should retry (I expect this to happen looking at the code).

To solve the firewall issue, I'd not encode the base URL in the code but use e.g. a `--release-url` parameter that allows to customize it (or something like that, even a template url). This way, the template can be provided in the local network in case of issues.

For 1.8 I think we should change the logic from "downloading the template", to "downloading a tar".. but I haven't looked carefully at the changes in the operator.

**Comment by lgarciaaco** [comment]
@zregvart I didn't change the `fileUrl`, I don't think it is necessary since it is used in only one place, and it will eventually be modified by adding the release version ...

**Comment by lgarciaaco** [comment]
fixes https://github.com/syndesisio/syndesis/issues/5925

**Pull request merged**